core/message_queue.cpp

```diff
@@ -268,7 +268,10 @@ void MessageQueue::flush() {
 	//using reverse locking strategy
 	_THREAD_SAFE_LOCK_
 
-	ERR_FAIL_COND(flushing); //already flushing, you did something odd
+	if (flushing) {
+		_THREAD_SAFE_UNLOCK_
+		ERR_FAIL_COND(flushing); //already flushing, you did something odd
+	}
```

Comment:
Maybe the test can be made before locking to prevent having to unlock? Not sure if it matters for the actual logic.

Reply:
`flushing` needs to be checked while the lock is held for thread safety. Moreover, the `ERR_FAIL_COND` is not following that so that code needs to be written another way.